tests/tests/test-precompile-assets-erc20.ts

```diff
@@ -0,0 +1,410 @@
+import { expect } from "chai";
+import { describeDevMoonbeam } from "../util/setup-dev-tests";
+import { customWeb3Request } from "../util/providers";
+import {
+  GENESIS_ACCOUNT,
+  ALITH,
+  BALTATHAR,
+  ALITH_PRIV_KEY,
+  CHARLETH,
+  BALTATHAR_PRIV_KEY,
+} from "../util/constants";
+import { blake2AsU8a, xxhashAsU8a } from "@polkadot/util-crypto";
+import { BN, hexToU8a, bnToHex, u8aToHex } from "@polkadot/util";
+import Keyring from "@polkadot/keyring";
+import { getCompiled } from "../util/contracts";
+import { ethers } from "ethers";
+import { createContract, createTransaction } from "../util/transactions";
+
+const sourceLocationRelay = { parents: 1, interior: "Here" };
+
+const sourceLocationRelayAssetType = { XCM: { parents: 1, interior: "Here" } };
+
+interface AssetMetadata {
+  name: string;
+  symbol: string;
+  decimals: BN;
+  isFrozen: boolean;
+}
+
+const relayAssetMetadata: AssetMetadata = {
+  name: "DOT",
+  symbol: "DOT",
+  decimals: new BN(12),
+  isFrozen: false,
+};
+
+async function mockAssetBalance(context, assetBalance, assetDetails, sudoAccount, assetId) {
```

Comment:
It would be great to move this in util for future tests

Reply:
Yes, I want to talk to @joelamouche for that

Reply:
@girazoki pls create ticket

Reply:
https://purestake.atlassian.net/browse/MOON-1130